### DISCUSSION TITLE
AugmentationSequential and ImageToTensor with a mask

### DISCUSSION THREAD

**Discussion by pietz** [kornia/kornia#1567] [Potential Issue]
I'm using the AugmentationSequential class for a image-mask-pair transformation pipeline. However, ImageToTensor doesn't seem to work with Datakey.MASK.

Is there a good reason for it or is it a potential feature that could be added?

**Comment by edgarriba** [comment]
@shijianjian

**Comment by shijianjian** [comment]
Currently, we do not suppport operations other than inside `Kornia.augmentation`. The major reason is that we cannot guarentee the behaviour of unkown functions. Say, what would `ImageToTensor` do to `bbox` and `keypoints`? A bit risky to allow that.

I was thinking of providing a preprocessing and postprocessing functionalities to support the needs like this. Say:
```python
AugmentationSequential(
    RandomAffine(),
    image_preproc=ImageToTensor(),
    mask_preproc=ImageToTensor(),
    image_postproc=TensorToImage(),
    mask_postproc=TensorToImage(),
)
``` 

Any comments?
